Detectors/CTP/simulation/src/Digits2Raw.cxx

```diff
@@ -0,0 +1,199 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// \file Digits2Raw.cxx
+/// \author Roman Lietava
+
+
+#include "CTPSimulation/Digits2Raw.h"
+#include "FairLogger.h"
+#include "CommonUtils/StringUtils.h"
+
+
+using namespace o2::ctp;
+
+//ClassImp(Digits2Raw);
+
+void Digits2Raw::init()
+{
+  // CTP Configuration
+    if(mCCDBServer.empty()) {
+    LOG(FATAL) << "CTP digitizer: CCDB server is not set";
+  } else {
+    LOG(INFO) << "CTP digitizer:: CCDB server:" << mCCDBServer;  
+  }
+  auto& mgr = o2::ccdb::BasicCCDBManager::instance();
+  mgr.setURL(mCCDBServer);
+  mCTPConfiguration = mgr.get<CTPConfiguration>(o2::ctp::CCDBPathCTPConfig);
+  //
+  // Register links
+  //
+  std::string outd = mOutDir;
+  if (outd.back() != '/') {
+    outd += '/';
+  }  
+  mLinkID = uint32_t(0);
+  mCruID = uint16_t(0);
+  mEndPointID = uint32_t(0);
+  for (int ilink = 0; ilink < mNLinks; ilink++) {
+    mFeeID = uint64_t(ilink);
+    std::string outFileLink = mOutputPerLink ? o2::utils::Str::concat_string(outd, "ctp_link", std::to_string(ilink), ".raw") : o2::utils::Str::concat_string(outd, "ctp.raw");
+    mWriter.registerLink(mFeeID, mCruID, mLinkID, mEndPointID, outFileLink);
```

Comment:
You should not have the same `mLinkID==0` on the same CRU. What is the point of this mLinkID at all if you know that you have 2 links?

Reply:
Fixed: mLinkID -> ilink

Reply:
On the git you still have ClassDefNV, could you push the code where you removed it and it still fails to compile?